src/analysis/functions.rs

```diff
@@ -465,6 +688,118 @@ fn analyze_async(
+                return None;
+            }
+            if let Some(destroy_index) = par.destroy_index {
+                if destroy_index >= c_parameters.len() {
+                    error!("function `{}` has an invalid destroy index of {} when there are {} parameters",
+                           func_name,
+                           destroy_index,
+                           c_parameters.len());
+                    return None;
+                }
+                if c_parameters[destroy_index].0.c_type != "GDestroyNotify" {
+                    *commented = true;
+                    error!("function `{}`'s callback `{}` has invalid destroy callback",
+                           func_name,
+                           par.name);
+                    return None;
+                }
+            }
+        }
+
+        // If we don't have a "user data" parameter, we can't get the closure so there's nothing we
+        // can do...
+        if par.c_type != "GDestroyNotify" &&
+           (func.parameters.is_empty() || !func.parameters.iter().any(|c| c.closure.is_some())) {
+            *commented = true;
+            error!("Closure type `{}` doesn't provide user data", par.c_type);
+            return None;
+        }
+
+        let parameters = ::analysis::trampoline_parameters::analyze(env, &func.parameters, par.typ, &[]);
+        if par.c_type != "GDestroyNotify" && !*commented {
+            *commented |= func.parameters.iter()
+                                         .any(|p| {
+                                             if p.closure.is_none() {
+                                                 ::analysis::trampolines::type_error(env, p).is_some()
+                                             } else {
+                                                 false
+                                             }
+                                         });
+        }
+        for p in parameters.rust_parameters.iter() {
+            if let Ok(s) = used_rust_type(env, p.typ, false) {
+                imports.add_used_type(&s, None);
+            }
+        }
+        if let Ok(s) = used_rust_type(env, func.ret.typ, false) {
+            if s != "GString" {
+                imports.add_used_type(&s, None);
+            } else {
+                imports.add_used_type(&"String", None);
+            }
+        }
+        Some((Trampoline {
+            name: par.name.to_string(),
+            parameters,
+            ret: func.ret.clone(),
+            bound_name: match callback_info {
+                Some(x) => x.bound_name,
+                None => return None,
+            },
+            bounds: Bounds::default(),
+            version: None,
+            inhibit: false,
+            concurrency: library::Concurrency::None,
+            is_notify: false,
+            scope: par.scope,
+            // If destroy callback, id doesn't matter.
+            user_data_index: c_parameters[par.user_data_index.unwrap_or_else(|| 0)].1,
```

Comment:
`unwrap_or_else` cause panic for glibs `g_log_set_writer_func` as `c_parameters` is empty
`index out of bounds: the len is 0 but the index is 0`